### PR TITLE
documentation: Fixes to command descriptions

### PR DESCRIPTION
**Pull request by balloonpopper**
Migration of https://github.com/godot-escoria/escoria-docs/pull/44

Outstanding questions below please (@StraToN @dploeger @BHSDuncan ) - once I've got clarification on the below I can make some further updates.

I've taken a stab at updating the ESC language commands documentation as there were things I thought could be worded better. Among other things, this fixes #124 - camera page links (#126 - camera limit documentation still needs a bit more general limits discussion, but I thought a future PR by someone for that would be better).

Some other pages as listed below I'm happy to update but I need your input on please.

HideMenu Command : hide_menu menu_type [enable_automatic_transition]
What is the difference between automatic_transition = true and = false? I don't understand what "automatic_transition" does.

QueueEvent :
I don't understand what channel is for, and the block documentation doesn't make sense to me either.

QueueResource:
As a user, I don't know what a "resource cache" is, or what a "resource" is, so I can't update the documentation to help other new users unfortunately.

RangGlobal:
It says both that the max_value is inclusive and exclusive, I need to know which it is to correct the documentation.

Repeat :
I'd like to give an example of where to use this to make it more obvious for users when this would come in handy.
Presumably if the event has a series of commands and you put a repeat at the end, the series repeats, but how do you break out of the loop? Is it possible?
Or do you only use this for something like animating a guard who walks to the left, stops, walks to the right, stops etc? How would I get him to stop so I could talk to him?

SchedEvent:
This implies that two events can't run at the same time as it queues the event if another is already running. Does this only apply to scheduled events - i.e. you can't run 2 scheduled events at the same time (the second queues), but you can run a scheduled event while normal scene events are in progress?

SetAngle:
The wait time isn't clear. "Number of seconds to wait for each animation occurring between the
current angle of object and the angle specified. "
What are the animations it plays on each rotation? The idle animation?

SetAnimations:
What's an animation resource? What format is the "animations" parameter (and how do I create one of these? - is it just an AnimatedSprite?)

SetInteractive:
What exactly does it mean to be interactive? How does a non-interactive object differ from a sprite that I put in my scene?

SetState:
It says that it "Can be used to change the appearance of an item or player character. See https://docs.escoria-framework.org/states for details."
The states link says nothing about changing the appearance of an item/player. If someone can give me a brief overview of how that works I'll update the documentation.

ShowMenu :
same as HideMenu - no idea what automatic_transition does

Stop:
What's the scope? If it's inside an if block, is it only the if block that is affected or does it terminate the event?

StopSnd :
Says "Stops the given sound player's stream" - not "stops the specified sound/speed/music file that is currently playing".
Does this mean you can only run 1 x _sound file at a time, 1 x _speech file at a time and 1 x _music file at at time?
This limitation (if it exists) is not listed in PlaySnd.

Transition
Can a user provide a transition? Are there requirements for the transition they must follow if they want to write their own?
Is a transition "in" just playing the shader backwards? (Or is the "out" just playing the shader backwards?)

turn_to:
Which animation(s) does it use for each turn? Is it playing the "idle" animation (and will it just play the first frame or the whole idle animation?)

walk_block:
Does this move the ESCPlayer/ESCItem to the coordinate of target? If the target has a ESCLocation does it go to the target or the ESCLocation?

Walk:
Speed as above

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd [hunk 1]
```diff
@@ -13,9 +15,9 @@
 # Supported transitions include the names of the values used
 # in the "TransitionType" enum of the "Tween" type (without the "TRANS_" prefix):
 #
-# https://docs.godotengine.org/en/stable/classes/class_tween.html?highlight=tween#enumerations
+# See https://docs.godotengine.org/en/stable/classes/class_tween.html?highlight=tween#enumerations
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]: @dploeger (Comment from @BHSDuncan on original PR) : will this tie us into this particular link always using the devel branch of the docs? What if we eventually deploy from a different branch? Or am I way off?

There are a few references pointing to this development branch link across this PR that may need to be corrected based on your response.

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+# Executes the specified Godot function. This function must be in a script
+# attached to a child node of a registered `ESCitem`.
 #
 # **Parameters**
 #
 # - *object*: Global ID of the target `ESCItem`
 # - *node*: Name of the child node of the target `ESCItem`
 # - *func_name*: Name of the function to be called
-# - *params*: Any primitive, non-array arguments for the function. Multiple
+# - *params*: Any non-array arguments to be passed to the function. Multiple
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: @dploeger : Comment from @BHSDuncan on this change - Do we allow for objects to be used as params @dploeger ? I thought we only allowed non-array primitive types?

File: addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd [hunk 3]
```diff
@@ -1,10 +1,14 @@
 # `hide_menu menu_type [enable_automatic_transition]`
 #
-# Hides either the main menu or the pause menu.
+# Hides either the main menu or the pause menu. The enable_automatic_transition
+# parameter can be used to specify if Escoria manages the graphical transition
+# or not. If set to false, you can perform your own transition as part of the
+# room's `setup` event (if you want to change the look of the transition used
+# from the default for example) using the `transition` command.
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd:7]: @StraToN - Comment from @BHSDuncan on my earlier PR : "I actually think you do it as part of both when exiting a room and when entering a room, so not just setup but also the exiting event in the previous room. @StraToN those correspond to transitioning out and in, no?"

File: addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd (hunk 1 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]: Yes, the way here is to go with redirects. /camera should be a redirect to the develop page currently, but apparently is missing.

That's a thing we can configure on the read the docs admin page.

Can you kindly filter out the revelant links and list here? I'll add the redirects then.

Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]: @dploeger Here's all the links listed based off a grep in every .gd file

https://docs.escoria-framework.org/camera
https://docs.escoria-framework.org/en/devel/getting_started/camera.html
https://docs.escoria-framework.org/states for details.
https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-match)
https://docs.godotengine.org/en/stable/classes/class_tween.html#enumerations
https://docs.godotengine.org/en/stable/classes/class_tween.html?highlight=tween#enumerations

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd (hunk 2 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: Yes. Agreed. No objects.

File: addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd (hunk 1 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]: - [ ] https://docs.escoria-framework.org/camera
- [ ] https://docs.escoria-framework.org/en/devel/getting_started/camera.html
- [ ] https://docs.escoria-framework.org/states
- [ ] https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-match
- [ ] https://docs.godotengine.org/en/stable/classes/class_tween.html#enumerations
- [ ] https://docs.godotengine.org/en/stable/classes/class_tween.html?highlight=tween#enumerations)

Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]: Okay, that's not much. You can use https://docs.escoria-framework.org/camera for https://docs.escoria-framework.org/en/devel/getting_started/camera.html.

I just added the redirect.

The godot docs are outside of our jurisdiction, so I'd just keep the pages under docs.godotengine.org as is.

Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd:20]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/camera_set_pos.gd [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+# Moves the camera to the given absolute position over a time period.
 #
 # **Parameters**
 #
 # - *speed*: Number of seconds the transition should take
 # - *x*: Target X coordinate
 # - "y*: Target Y coordinate
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_set_pos.gd:11]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/camera_set_target.gd [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
 #
 # **Parameters**
 #
-# - *speed*: Number of seconds the transition should take
+# - *speed*: Number of seconds the transition should take to move the camera 
+#   to follow `object`
 # - *object*: Global ID of the target object
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_set_target.gd:13]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/camera_set_zoom.gd [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
+# would result in no change. The zoom will happen over the given time period.
 #
 # **Parameters**
 #
 # - *magnitude*: Magnitude of zoom
 # - *time*: Number of seconds the transition should take, with a value of `0`
 #   meaning the zoom should happen instantly (default: `0`)
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_set_zoom.gd:15]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/camera_set_zoom_height.gd [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+# Zooms the camera in/out so it occupies the given height in pixels.
 #
 # **Parameters**
 #
 # - *pixels*: Target height in pixels
 # - *time*: Number of seconds the transition should take, with a value of `0`
 #   meaning the zoom should happen instantly (default: `0`)
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_set_zoom_height.gd:11]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/camera_shift.gd [hunk 8]
```diff
@@ -15,7 +16,7 @@
 #
 # https://docs.godotengine.org/en/stable/classes/class_tween.html?highlight=tween#enumerations
 #
-# For more details see: https://docs.escoria-framework.org/camera
+# For more details see: https://docs.escoria-framework.org/en/devel/getting_started/camera.html
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/camera_shift.gd:19]:
```suggestion
# For more details see: https://docs.escoria-framework.org/camera
```

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd (hunk 2 above)
Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: So how would you like this reworded please @dploeger ?

File: addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd (hunk 1 above)
Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/camera_push.gd]: Thanks @dploeger - I've corrected all the camera references accordingly (I'll squash the commits at a later point)

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd (hunk 2 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: Hmmmm. Maybe like this?

```
# - *params*: Arguments to the function (array and object parameters are not supported). (...)
```

Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: "Any primitive, non-array argument" would include object, right? Or is an object non-primitive? Maybe. It's really nitpicky I'd say. :)

Comment (BHSDuncan) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: primitive = non-object, non-array (usually native/intrinsic data types to the compiler, typically int, float, double, etc., and usually string, but I guess not always).

All good, though.

Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: That was the reason I removed the word "primative". I didn't think the term was accessible enough to the audience that they would be sure of the meaning. As the developers are confirming here exactly what it entails, it shows me it's not the right word to use - if it's not 100% immediately clear to you, then the users are going to be totally lost.

If you would like to leave the intent, can I suggest a variant of @dploeger's version from 9 hours ago? 
# - *params*:  Any arguments to be passed to the function (array and object parameters are not supported).

Comment (BHSDuncan) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]: I'm good with that.

I take your point; I sometimes forget that not everyone who's going to use this is a software engineer or someone who just inherently/intuitively knows what "primitive" means/represents.

Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc/commands/custom.gd:12]:
```suggestion
# - params: Any arguments to be passed to the function (array and object parameters are not supported). Multiple
```

**Comment by BHSDuncan** [comment]
Can't remember if we came to a decision on this, but we should for consistency's sake:

Are we going with `dialog` or 'dialogue` throughout the docs? We really should pick one and stick with it.

@dploeger @StraToN

**Comment by balloonpopper** [comment]
I wasn't sure so I left it at as-is. I'll update all of them for consistency based on the feedback (there's currently 2 x "dialogue" and the rest are "dialog"  😬 )

**Comment by BHSDuncan** [comment]
> I wasn't sure so I left it at as-is. I'll update all of them for consistency based on the feedback (there's currently 2 x "dialogue" and the rest are "dialog" 😬 )

Yeah no worries. This is something I'd meant to address awhile ago but it kinda slipped through the cracks. Appreciate all your work!

**Comment by StraToN** [comment]
> Can't remember if we came to a decision on this, but we should for consistency's sake:
> 
> Are we going with `dialog` or 'dialogue` throughout the docs? We really should pick one and stick with it.
> 
> @dploeger @StraToN

After checking around for a confirmation of my assumption of the usage of both words, here is what I think we should use:

- "dialog" refers to GUI element, such as "popup dialog" or "dialog box"
- "dialogue" refers to the conversational meaning

This mean that a "dialog tree" is not correct, we should use "dialogue tree" in this case, and so on. WDYT?

File: addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd (hunk 3 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd:7]: Yes, that's exactly it. 
If you disable automatic transitions, then you have to manage the transition out as you exit the scene (so in the :exit event, for an ESCExit) *and* you also have to manage the transition in the new room (at the end of the :setup event of the new room, usually)

**Review by StraToN — APPROVED** [review]

File: addons/escoria-core/game/core-scripts/esc/commands/turn_to.gd [hunk 9]
```diff
@@ -2,6 +2,12 @@
 #
 # Turns `object` to face another object.
 #
+# Unlike movement commands, `turn_to` will not automatically reference an
+# ESClocation that is a child of an ESCitem. 
+# To turn towards a ESClocation that is a child of an ESCitem, give that 
+# ESClocation a `Global ID`, and set the `object_to_face` parameter to this 
+# `Global ID`.
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc/commands/turn_to.gd:9]:
```suggestion
# Unlike movement commands, `turn_to` will not automatically reference an
# `ESCLocation` that is a child of an `ESCItem.` 
# To turn towards an `ESCLocation` that is a child of an `ESCItem`, give the
# `ESCLocation` a `Global ID` and use this value as the `object_to_face`
# parameter.
```

File: addons/escoria-core/game/core-scripts/esc_location.gd [hunk 10]
```diff
@@ -1,5 +1,8 @@
 # A simple node extending Position2D with a global ID so that it can be
-# referenced in ESC Scripts.
+# referenced in ESC Scripts. Movement-based commands like walk_to_pos will
+# automatically use an ESClocation that is a child of the destination node.
+# Commands like turn_to which are not movement-based will ignore child
+# ESClocations and refer to the parent node.
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_location.gd:5]:
```suggestion
# referenced in ESC Scripts. Movement-based commands like `walk_to_pos` will
# automatically use an `ESCLocation` that is a child of the destination node.
# Commands like `turn_to`--which are not movement-based--will ignore child
# `ESCLocation`s and refer to the parent node.
```

File: addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd (hunk 3 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd:7]: @balloonpopper Does this answer your question? May I mark the discussion as resolved?

Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd:7]: @StraToN - I'll need to update the text as I only mentioned transition in. 
If I'm reading the code correctly, it appears the ":exit" event is actually ":exit_scene". Can you please confirm and then I'll update the documentation to add
"Transitions when leaving a room can be managed manually as part of the `exit_scene` event."

**Review by StraToN — APPROVED** [review]

File: addons/escoria-core/game/core-scripts/esc_location.gd (hunk 10 above)
Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc_location.gd:5]: I'm not sure how "--" gets translated by the document generator, but I'll take your word for it that that's how it should look @BHSDuncan 😄

File: addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd (hunk 3 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd:7]: Correct, the actual event name is `exit_scene`, not `exit`.

**Review by BHSDuncan — APPROVED** [review]

File: addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd (hunk 3 above)
Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc/commands/hide_menu.gd]: I'll leave updating the documentation for hide_menu for now. I want to clarify when you'd expect to use it (see [issue 179](https://github.com/godot-escoria/escoria-issues/issues/179) ) and will modify the text accordingly.

**Comment by StraToN** [comment]
Thanks!

**Pull request merged**